gwpy/timeseries/tests/test_timeseries.py

```diff
@@ -924,12 +924,19 @@ def test_notch(self, losc):
             losc.notch(10, type='fir')
 
     def test_q_transform(self, losc):
-        # test simple q-transform
+        # test simple q-transform, with linear and log frequency spacing
         qspecgram = losc.q_transform(method='scipy-welch', fftlength=2)
+        qspecgram_logf = losc.q_transform(method='scipy-welch', fftlength=2,
+                                          fres=500, logf=True)
         assert isinstance(qspecgram, Spectrogram)
+        assert isinstance(qspecgram_logf, Spectrogram)
         assert qspecgram.shape == (4000, 2403)
+        assert qspecgram_logf.shape == (4000, 500)
         assert qspecgram.q == 5.65685424949238
-        nptest.assert_almost_equal(qspecgram.value.max(), 156.0411964254892)
+        assert qspecgram_logf.q == 5.65685424949238
+        nptest.assert_almost_equal(qspecgram.value.max(), 156.43279233248313)
+        nptest.assert_almost_equal(qspecgram_logf.value.max(),
+                                   156.43222488296405)
```

Comment:
@alurban, can you refactor this into two separate tests, something like `test_q_transform` and `test_q_transform_logf`? I think this will be cleaner to maintain.

Reply:
@duncanmmacleod, that's a good idea, I did as you suggested and defined a separate unit test called `test_q_transform_logf`.